relayer/message_relayer.go

```diff
@@ -125,8 +144,79 @@ func (r *messageRelayer) relayMessage(warpMessageInfo *vmtypes.WarpMessageInfo,
 	return nil
 }
 
-// Run collects signatures from nodes by directly querying them via AppRequest, then aggregates the signatures, and constructs the signed warp message.
-func (r *messageRelayer) createSignedMessage(requestID uint32) (*warp.Message, error) {
+func (r *messageRelayer) createSignedMessage() (*warp.Message, error) {
+	r.logger.Info("Fetching aggregate signature from the source chain validators")
+	warpClient, err := warpBackend.NewClient(r.relayer.apiNodeURI, r.relayer.sourceBlockchainID.String())
+	if err != nil {
+		r.logger.Error(
+			"Failed to create warp client",
+			zap.Error(err),
+		)
+		return nil, err
+	}
+	signingSubnetID := r.relayer.sourceSubnetID
+	if r.relayer.sourceSubnetID == constants.PrimaryNetworkID {
+		signingSubnetID, err = r.relayer.pChainClient.ValidatedBy(context.Background(), r.destinationBlockchainID)
+		if err != nil {
+			r.logger.Error(
+				"failed to get validating subnet for destination chain",
+				zap.String("destinationBlockchainID", r.destinationBlockchainID.String()),
+				zap.Error(err),
+			)
+			return nil, err
+		}
+	}
+
+	var signedWarpMessageBytes []byte
+	for attempt := 1; attempt <= maxRelayerQueryAttempts; attempt++ {
```

Comment:
```suggestion
	for attempt := 0; attempt < maxRelayerQueryAttempts; attempt++ {
```

Minor nit for the other folks who haven't worked at MathWorks 😅 


** Edit: Ah nevermind I see why you did this to make the "last iteration" check simpler below. I think in the past we've typically done -1 on the last iteration check rather than 1 indexing to start with, but no strong preference.

Reply:
I would consider creating a "request with retries" `utils` method either way so that we can re-use this logic though.

Reply:
Discussed offline, we'll defer the retry utility for now because it's more complicated than it's worth for the time being.